File: frontend/src/lib/ordenDelDia.ts
import api from "@/lib/api"
import type { Objective } from "@/lib/annualPlan"

export interface ThemeRef {
  key: string
  label: string
  every_n_sessions: number | null
}

export interface OrdenDelDia {
  month_index: number
  period_year: number
  period_month: number
  permanent_themes: ThemeRef[]
  coverage_themes: ThemeRef[]
  covered_keys: string[]
  objectives: Objective[]
}

export async function getOrdenDelDia(monthIndex: number): Promise<OrdenDelDia> {
  const r = await api.get<OrdenDelDia>(`/annual-plan/months/${monthIndex}/orden-del-dia`)
  return r.data
}

export async function markCoverage(monthIndex: number, themeKey: string, covered: boolean): Promise<void> {
  await api.post(`/annual-plan/months/${monthIndex}/coverage`, { theme_key: themeKey, covered })
}

File: backend/app/schemas/orden_del_dia.py
from pydantic import BaseModel, Field

from app.schemas.annual_plan import ObjectiveOut


class ThemeRef(BaseModel):
    key: str
    label: str
    every_n_sessions: int | None = None


class OrdenDelDiaOut(BaseModel):
    month_index: int
    period_year: int
    period_month: int
    permanent_themes: list[ThemeRef] = Field(default_factory=list)
    coverage_themes: list[ThemeRef] = Field(default_factory=list)
    covered_keys: list[str] = Field(default_factory=list)
    objectives: list[ObjectiveOut] = Field(default_factory=list)
